packages/web/src/components/user-list-modal/components/UserListModalV2.tsx

```diff
@@ -0,0 +1,220 @@
+import { useRef } from 'react'
+
+import {
+  Modal,
+  Scrollbar,
+  IconTipping as IconTip,
+  IconUser,
+  IconUserGroup,
+  IconTrophy,
+  IconUserFollowing as IconFollowing,
+  IconCart,
+  IconRemix
+} from '@audius/harmony'
+import { ChatBlastAudience } from '@audius/sdk'
+import { useDispatch, useSelector } from 'react-redux'
+import { useRouteMatch } from 'react-router-dom'
+
+import { ChatBlastWithAudienceCTA } from 'pages/chat-page/components/ChatBlastWithAudienceCTA'
+import {
+  getUserListType,
+  getIsOpen
+} from 'store/application/ui/userListModal/selectors'
+import { setVisibility } from 'store/application/ui/userListModal/slice'
+import { UserListType } from 'store/application/ui/userListModal/types'
+
+import styles from './UserListModal.module.css'
+import { FavoritesUserList } from './lists/FavoritesUserList'
+import { FollowersUserList } from './lists/FollowersUserList'
+import { FollowingUserList } from './lists/FollowingUserList'
+import { MutualsUserList } from './lists/MutualsUserList'
+import { NotificationUserList } from './lists/NotificationUserList'
+import { PurchasersUserList } from './lists/PurchasersUserList'
+import { RelatedArtistsUserList } from './lists/RelatedArtistsUserList'
+import { RemixersUserList } from './lists/RemixersUserList'
+import { RepostsUserList } from './lists/RepostsUserList'
+import { SupportingUserList } from './lists/SupportingUserList'
+import { TopSupportersUserList } from './lists/TopSupportersUserList'
+
+const messages = {
+  reposts: 'Reposts',
+  favorites: 'Favorites',
+  followers: 'Followers',
+  following: 'Following',
+  topSupporters: 'Top Supporters',
+  supporting: 'Supporting',
+  relatedArtists: 'Related Artists',
+  mutuals: 'Mutuals',
+  purchasers: 'Purchasers',
+  remixers: 'Remixers'
+}
+
+const UserListModalV2 = () => {
+  const dispatch = useDispatch()
+  const userListType = useSelector(getUserListType)
+  const isOpen = useSelector(getIsOpen)
+  const onClose = () => dispatch(setVisibility(false))
+  const scrollParentRef = useRef<HTMLElement>()
+
+  const match = useRouteMatch<{ audience_type: string }>(
+    '/messages/:audience_type'
+  )
+  const isChatBlastPath =
+    match?.params.audience_type &&
+    Object.values(ChatBlastAudience).includes(
+      match.params.audience_type as ChatBlastAudience
+    )
+
+  const getUserList = () => {
+    switch (userListType) {
+      case UserListType.FAVORITE:
+        return {
+          component: <FavoritesUserList onClose={onClose} />,
+          title: messages.favorites
+        }
+      case UserListType.REPOST:
+        return {
+          component: <RepostsUserList onClose={onClose} />,
+          title: messages.reposts
+        }
+      case UserListType.FOLLOWER:
+        return {
+          component: (
+            <FollowersUserList
+              onClose={onClose}
+              getScrollParent={() => scrollParentRef.current || null}
```

Comment:
why does only follower need this?

Reply:
they all need it actually, so will figure out a better strat when i implement the rest